community/io/src/main/java/org/neo4j/io/fs/DefaultFileSystemAbstraction.java

```diff
@@ -231,6 +220,39 @@ public void deleteFileOrThrow( File file ) throws IOException
         return StreamFilesRecursive.streamFilesRecursive( directory, this );
     }
 
+    @Override
+    public void forceDirectory( Path dir )
+    {
+        // Lucene relies on flushing directory metadata to disk for commit safety; this code is from their way of doing that.
```

Comment:
Agree this is kind of.. broken. Clearly agree Lucene shouldn't be mentioned here - but at the same time, hard for future readers to really see context without talking about it, since it's the only reason it's here :/ I'm modifying to link to the JDK ticket and not talk directly about Lucene.

On need for this: As you noticed further down, we only read via the Page Cache; reasons for why added to that comment below. That means that, at least for now, we can't flush via the page cache. 

I guess this answers your earlier question about files and file formats: This uses (for now, at least) the *exact* same disk format as regular Lucene, and the *exact* same set of files. It is the exact same write path. The only difference is that we do reads via the Page Cache instead of memory mapping.